src/pip/_internal/vcs/git.py

```diff
@@ -74,27 +74,41 @@ def export(self, location):
                 show_stdout=False, cwd=temp_dir.path
             )
 
-    def check_rev_options(self, rev, dest, rev_options):
-        """Check the revision options before checkout to compensate that tags
-        and branches may need origin/ as a prefix.
-        Returns the SHA1 of the branch or tag if found.
+    def get_revision_sha(self, dest, rev):
         """
-        revisions = self.get_short_refs(dest)
-
-        origin_rev = 'origin/%s' % rev
-        if origin_rev in revisions:
-            # remote branch
-            return [revisions[origin_rev]]
-        elif rev in revisions:
-            # a local tag or branch name
-            return [revisions[rev]]
-        else:
+        Return a commit hash for the given revision if it names a remote
+        branch or tag.  Otherwise, return None.
+
+        Args:
+          dest: the repository directory.
+          rev: the revision name.
+        """
+        # Pass rev to pre-filter the list.
+        output = self.run_command(['show-ref', rev], cwd=dest,
+                                  show_stdout=False, on_returncode='ignore')
+        refs = {}
+        for line in output.strip().splitlines():
+            sha, ref = line.split(None, 1)
```

Comment:
Is there an edge case I'm missing here? I don't think refs can contain whitespace characters (see [`check-ref-format`](https://git-scm.com/docs/git-check-ref-format)), so it seems that `line.split()` should suffice. ([Perhaps it is just historical](https://github.com/pypa/pip/commit/92e0dbbf4ceeba7f4c4d736c276447b70d4732ed#diff-4cb54fa0d0d77f2cf5e44310056fec4aR146))

Reply:
Thanks for reviewing! I did that to make fewer assumptions about the output, and to get a more informative error in the off-chance there is extra whitespace. We would see the parsed ref as opposed to the less user-friendly, "ValueError: too many values to unpack (expected 2)". I also feel this is better than slicing off the first two elements, which could silently mask issues.